### PR TITLE
 Fixing shellcheck warnings in prin

### PR DESCRIPTION
**Pull request by pswica**
Please see: https://github.com/kubernetes/release/pull/742 and https://github.com/kubernetes/release/issues/726 for more details and context.

Let me know anything else to improve on :)

### CODE REVIEW

File: prin [hunk 1]
```diff
@@ -142,15 +144,15 @@ if [[ ${POSITIONAL_ARGV[0]} =~ [0-9]{5,} ]]; then
   # non-cherrypick commit searches on branches
   INITIAL_COMMIT=$(git log --grep "Merge pull request #$PR" \
                            --all --pretty=format:"%h")
-  CP_COMMITS=($(git log --grep "cherry-pick-of-.*#$PR-" \
-                        --all --pretty=format:"%h"))
+  mapfile -i CP_COMMITS < <(git log --grep "cherry-pick-of-.*#$PR-" \
+                        --all --pretty=format:"%h")
```
Comment (xauthulei) [inline prin]: @pswica, here I met this error when used `mapfile  -i` :
`/Users/llhu/Desktop/test/release/release/prin: line 154: mapfile: -i: invalid option
mapfile: usage: mapfile [-d delim] [-n count] [-O origin] [-s count] [-t] [-u fd] [-C callback] [-c quantum] [array]`
Is there some errors in my local side. Thanks

Comment (pswica) [reply prin]: Thanks os much for catching. No this was my bad. I messed up by putting `-i` instead of `-t`

Comment (pswica) [reply prin]: PR here

File: prin [hunk 2]
```diff
@@ -70,7 +72,7 @@ show_tags () {
   local pr=$1
   local initial_commit=$2
   shift 2
-  local cp_commits=($*)
+  local cp_commits=("$*")
```
Comment (spiffxp) [inline prin]: There is a good chance this should be `"$@"` instead, ref: https://github.com/koalaman/shellcheck/wiki/SC2086#exceptions

File: prin [hunk 3]
```diff
@@ -142,15 +144,15 @@ if [[ ${POSITIONAL_ARGV[0]} =~ [0-9]{5,} ]]; then
...
   INITIAL_COMMIT=("${POSITIONAL_ARGV[0]}")
 fi
 
-[[ -z "$INITIAL_COMMIT" ]] && common::exit 1 "No commit/pr found in this repo."
+[[ -z "${INITIAL_COMMIT[0]}" ]] && common::exit 1 "No commit/pr found in this repo."
 
 security_layer::hosted_map
 
-show_tags "$PR" $INITIAL_COMMIT ${CP_COMMITS[*]}
+show_tags "$PR" "${INITIAL_COMMIT[0]}" "${CP_COMMITS[*]}"
```
Comment (spiffxp) [inline prin]: There is a good chance this should be `"${CP_COMMITS[@]}"` instead, ref: https://github.com/koalaman/shellcheck/wiki/SC2086#exceptions

Comment (pswica) [reply prin]: Thanks a lot for reading these. I'm sure there is a lot of hidden bugs/weirdness from my shellcheck changes. I will create an issue and link this, and other things found, to it

**Pull request merged**